PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'maintenance.equipment'
-    _inherit = 'maintenance.request'
-    _inherit = 'res.users'
-    _inherit = 'hr.employee'

Total Changes: 12

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/hr_maintenance/models/equipment.py
+++ b/addons/hr_maintenance/models/equipment.py
@@ -4,7 +4,7 @@ from odoo import api, fields, models, tools
 
 
 class MaintenanceEquipment(models.Model):
-    _inherit = 'maintenance.equipment'
+    _inherit = ['maintenance.equipment']
 
     employee_id = fields.Many2one('hr.employee', compute='_compute_equipment_assign',
         store=True, readonly=False, string='Assigned Employee', tracking=True)
@@ -78,7 +78,7 @@ class MaintenanceEquipment(models.Model):
 
 
 class MaintenanceRequest(models.Model):
-    _inherit = 'maintenance.request'
+    _inherit = ['maintenance.request']
 
     @api.returns('self')
     def _default_employee_get(self):

--- a/addons/hr_maintenance/models/res_users.py
+++ b/addons/hr_maintenance/models/res_users.py
@@ -1,8 +1,8 @@
 from odoo import api, models, fields
 
 
-class Users(models.Model):
-    _inherit = 'res.users'
+class ResUsers(models.Model):
+    _inherit = ['res.users']
 
     equipment_ids = fields.One2many('maintenance.equipment', 'owner_user_id', string="Managed Equipment")
     equipment_count = fields.Integer(related='employee_id.equipment_count', string="Assigned Equipment")
@@ -12,8 +12,8 @@ class Users(models.Model):
         return super().SELF_READABLE_FIELDS + ['equipment_count']
 
 
-class Employee(models.Model):
-    _inherit = 'hr.employee'
+class HrEmployee(models.Model):
+    _inherit = ['hr.employee']
 
     equipment_ids = fields.One2many('maintenance.equipment', 'employee_id', groups="hr.group_hr_user")
     equipment_count = fields.Integer('Equipment Count', compute='_compute_equipment_count', groups="hr.group_hr_user")
